js/app.js

```diff
@@ -0,0 +1,128 @@
+// Array to store currency rates
+let currencyRates = [];
+
+// Function for adding a new exchange rate
+function handleAddRateForm(event) {
+  event.preventDefault(); // Prevent the default form submission behavior
+
+  // Get the value of the base currency from the input field
+  const baseCurrency = document.getElementById("base-currency").value;
+
+  // Get the value of the target currency from the input field
+  const targetCurrency = document.getElementById("target-currency").value;
+
+  // Get the value of the exchange rate from the input field and convert it to a float
+  const exchangeRate = parseFloat(
+    document.getElementById("exchange-rate").value
+  );
+
+  // Find the rate object for the base currency, or create a new one if it doesn't exist
+  let rateObject = currencyRates.find((rate) => rate.base === baseCurrency);
+
+  if (!rateObject) {
+    // If no rate object is found for the base currency
+    rateObject = {
+      // Create a new rate object
+      timestamp: Date.now(), // Current timestamp
+      base: baseCurrency, // Set the base currency
+      date: new Date().toISOString().split("T")[0], // Current date in YYYY-MM-DD format
+      rates: {}, // Initialize an empty rates object
+    };
+    currencyRates.push(rateObject); // Add the new rate object to the currencyRates array
+  }
+
+  // Add or update the exchange rate for the target currency
+  rateObject.rates[targetCurrency] = exchangeRate;
+
+  // Log the current exchange rates array to the console
+  console.log("Current exchange rates:", currencyRates);
```

Comment:
It would be nice to display a success or error message to the user. I know how to open the browser console but John Doe working in a random bank does not.

Reply:
Added error and success messages with timer